docs/spec/distributing.rst

```diff
@@ -10,72 +10,428 @@ Stub files
+generally possible and recommended::
 
-* Extension modules
+    from collections.abc import Iterable
 
-* Third-party modules whose authors have not yet added type hints
+    def foo(iter: Iterable[int]) -> None: ...
 
-* Standard library modules for which type hints have not yet been
-  written
+Unions
+""""""
+
+Declaring unions with the shorthand syntax or ``Union`` and ``Optional`` is
+supported by all type checkers::
+
+    def foo(x: int | str) -> int | None: ...  # recommended
+    def foo(x: Union[int, str]) -> Optional[int]: ...  # ok
+
+Module Level Attributes
+"""""""""""""""""""""""
+
+Module level variables and constants can be annotated using either
+type comments or variable annotation syntax::
+
+    x: int  # recommended
+    x: int = 0
+    x = 0  # type: int
+    x = ...  # type: int
+
+The ellipsis literal can stand in for any value::
+
+    x: int = ...  # type is int
+
+A variable annotated as ``Final`` and assigned a literal value has the
+corresponding ``Literal`` type::
+
+    x: Final = 0  # type is Literal[0]
+
+In all other cases, the type of a variable is unspecified when the variable is
+unannotated or when the annotation and the assigned value disagree::
+
+    x = 0  # type is unspecified
+    x = ...  # type is unspecified
+    x: int = ""  # type is unspecified
+
+Classes
+"""""""
+
+Class definition syntax follows general Python syntax, but type checkers
+are only expected to understand the following constructs in class bodies:
+
+* The ellipsis literal ``...`` is ignored and used for empty
+  class bodies. Using ``pass`` in class bodies is undefined.
+* Instance attributes follow the same rules as module level attributes
+  (see above).
+* Method definitions (see below) and properties.
+* Method aliases.
+* Inner class definitions.
+
+Yes::
+
+    class Simple: ...
+
+    class Complex(Base):
+        read_write: int
+        @property
+        def read_only(self) -> int: ...
+        def do_stuff(self, y: str) -> None: ...
+        doStuff = do_stuff
+        class Inner: ...
+
+More complex statements don't need to be supported.
+
+The type of generic classes can be narrowed by annotating the ``self``
+argument of the ``__init__`` method::
+
+    class Foo(Generic[_T]):
+        @overload
+        def __init__(self: Foo[str], type: Literal["s"]) -> None: ...
+        @overload
+        def __init__(self: Foo[int], type: Literal["i"]) -> None: ...
+        @overload
+        def __init__(self, type: str) -> None: ...
+
+The class must match the class in which it is declared. Using other classes,
+including sub or super classes, will not work. In addition, the ``self``
+annotation cannot contain type variables.
+
+Functions and Methods
+"""""""""""""""""""""
 
-* Modules that must be compatible with Python 2 and 3
+Function and method definition syntax follows general Python syntax.
+For backwards compatibility, positional-only parameters can also be marked by
```

Comment:
The information about positional-only parameters is redundant (already covered elsewhere in the spec) and should not be repeated in this section.